perun-core/src/main/java/cz/metacentrum/perun/core/impl/GroupsManagerImpl.java

```diff
@@ -129,21 +131,28 @@ public Group createGroup(PerunSession sess, Vo vo, Group group) throws GroupExis
 			throw new InternalErrorException(e);
 		}
 
+		int newId;
 		try {
 			// Store the group into the DB
-			int newId = Utils.getNewId(jdbc, "groups_id_seq");
+			newId = Utils.getNewId(jdbc, "groups_id_seq");
 
 			jdbc.update("insert into groups (id, parent_group_id, name, dsc, vo_id, created_by,created_at,modified_by,modified_at,created_by_uid,modified_by_uid) " +
 					"values (?,?,?,?,?,?," + Compatibility.getSysdate() + ",?," + Compatibility.getSysdate() + ",?,?)", newId, group.getParentGroupId(),
 					group.getName(), group.getDescription(), vo.getId(), sess.getPerunPrincipal().getActor(), sess.getPerunPrincipal().getActor(), sess.getPerunPrincipal().getUserId(), sess.getPerunPrincipal().getUserId());
-			group.setId(newId);
-
-			group.setVoId(vo.getId());
-
-			return group;
 		} catch (RuntimeException err) {
 			throw new InternalErrorException(err);
 		}
+		Group newGroup;
+		try {
+			newGroup = getGroupById(sess, newId);
+		} catch (GroupNotExistsException e) {
+			throw new InternalErrorException("Failed to read newly created group with id: " + newId, e);
+		}
+		group.setId(newId);
+		group.setUuid(newGroup.getUuid());
+		group.setVoId(newGroup.getVoId());
```

Comment:
I don't really think this is necessary.  You get the whole new group correctly set with all expected attributes.

Reply:
Actually, it is. Because a lot of code expects the given object to be modified. Without it, a lot of code would have to be rewritten.

Reply:
Ok so please just comment it.

Reply:
Regarding this, don't we want to then return ``group`` object instead of the ``newGroup`` ?

And just a note, it would be so nice to use Postgres ``INSERT .... RETURNING`` query, but its not supported by HSQLDB which we use for the tests.